Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,242 @@
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Institue of Cancer Research.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Guillaume GODIN access the AutoCorrelation 3D descriptors names in Dragon TDB
+
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+#include "AUTOCORR3D.h"
+#include "MolData3Ddescriptors.h"
+
+#include "GraphMol/PartialCharges/GasteigerCharges.h"
+#include "GraphMol/PartialCharges/GasteigerParams.h"
+#include <Numerics/EigenSolvers/PowerEigenSolver.h>
+
+#include <Numerics/Matrix.h>
+#include <Numerics/SquareMatrix.h>
+#include <Numerics/SymmMatrix.h>
+#include <boost/foreach.hpp>
+#include <math.h>
+#include <Eigen/Dense>
+#include <Eigen/SVD>
+#include <iostream>
+#include <Eigen/Core>
+#include <Eigen/QR>
+
+
+
+using namespace Eigen;
+namespace RDKit {
+namespace Descriptors{
+
+namespace {
+
+MolData3Ddescriptors moldata3D;
+
+double* retreiveMat(MatrixXd matrix) {
+     double* arrayd = matrix.data();
+     return arrayd;
+  }
+
+double* retreiveVect(VectorXd matrix) {
+     double* arrayd = matrix.data();
+     return arrayd;
+
+  }
+
+VectorXd getEigenVect(std::vector<double> v){
+
+      double* varray_ptr = &v[0];
+      Map<VectorXd> V(varray_ptr,v.size());
+      return V;
+  }
+
+
+double* AppendDouble(double *w, double* Append, int length, int pos){
+    for (int i=pos;i<pos+length;i++) {
+          w[i]=Append[i-pos];
+        }
+
+    return w;
+}
+
+
+double* GetGeodesicMatrix(double* dist, int lag,int numAtoms){
+    int sizeArray=numAtoms*numAtoms;
+    double *Geodesic = new double[sizeArray];
+    for (int i=0; i<sizeArray;i++) {
+      if (dist[i]==lag) Geodesic[i]=1;
+      else  Geodesic[i]=0;
```

Comment:
The RDkit adopted coding standard (https://google.github.io/styleguide/cppguide.html) prohibits having an 
> IF statement on one line when there is an ELSE clause